core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -108,6 +110,19 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
   val failedFetchRequestRate = newMeter("FailedFetchRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
   val totalProduceRequestRate = newMeter("TotalProduceRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
   val totalFetchRequestRate = newMeter("TotalFetchRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
+
+  val partitions = new ConcurrentHashMap[Int, Partition]
```

Comment:
@jjkoshy Sorry I had missed your note. I used a map instead of a counter to ensure that the synchronization around partition add and delete match that in ReplicaManager where the partitions are maintained. Perhaps this is not strictly necessary for metrics, but it does ensure that metrics of partitions that are deleted and immediately recreated works without losing any data. Metrics are deleted strictly once for the partition objects in ReplicaManager when a partition is deleted.